network/default.go

```diff
@@ -681,14 +706,20 @@ func (n *network) processCtrlChan(listener tunnel.Listener) {
 						Metric:  int(event.Route.Metric),
 					}
 					// set the route metric
-					n.node.RLock()
-					n.setRouteMetric(&route)
-					n.node.RUnlock()
-					// throw away metric bigger than 1000
-					if route.Metric > 1000 {
-						log.Debugf("Network route metric %d dropping node: %s", route.Metric, route.Router)
-						continue
+					n.RLock()
+					if link, ok := n.peerLinks[event.Route.Gateway]; ok {
+						// NOTE: should we change router.Route.Metric to int64?
+						if int(link.Length()) < route.Metric {
+							route.Metric = int(link.Length())
```

Comment:
Should we be doing more here besides just using link length? Maybe we want metric to still be a number between 1 and 1000 so it's readable. Also you want to check if length is 0 because it means it has not been measured yet.

Reply:
Yes, we totally should! This is a placeholder for the time being so we have a place to discuss.

Reply:
```
if delay == 0 {
  delay = 1
}

metric := (delay * length * hop) / 10e9

route.Metric += metric
```